### ISSUE TITLE
Allow binary blobs

### ISSUE THREAD

**Issue by hannesl** [github-tools/github#23]
In order to safely transfer binary blobs/filtes, the API needs to use base64 encoding when appropriate. postBlob() however always uses "utf-8". So there needs to be either an additional argument in write() and postBlob() that specifies the encoding, or postBlob() needs to detect if the file is binary or not, and adjust the encoding automatically.

This little snippet seems to be able to detect binary data:

```
/[\x00-\x08\x0E-\x1F]/.test(data)
```

See http://stackoverflow.com/questions/1677644/detect-non-printable-characters-in-javascript

I'm currently dealing with similar issues in Prose: https://github.com/prose/prose/pull/255

**Comment by aendrew** [comment]
@mattpass Was this ever dealt with? There seems to be some binary file handling in postBlob, which I'm looking at briefly due to #149.

**Comment by mattpass** [comment]
@aendrew Nope, was awaiting a pull request and going to check that out. Nothing submitted or coded by myself here.
